### PR TITLE
E2E Test For Hook-Enabled ERC-1271 Signatures

### PR DESCRIPTION
**Pull request by nlordell**
Now, thanks to #1831, we no longer depend on `eth_call` with state override support for simulating ERC-1271 signature validation for orders with pre-hooks. This means that we can finally write an E2E test for this!

This PR does just that - it executes an order on behalf of a Safe that is created just-in-time for the settlement as part of a pre-hook. Note that we assert that the ERC-1271 order is created for a Safe with no code!

### Test Plan

This is the test!

### CODE REVIEW

File: crates/e2e/src/setup/onchain_components.rs [hunk 1]
```diff
@@ -261,6 +269,106 @@ impl Deref for CowToken {
...
+    /// Returns the address of the Safe.
+    pub fn address(&self) -> H160 {
+        self.contract.address()
+    }
+
+    /// Returns the domain separator for the Safe.
+    fn domain_separator(&self) -> DomainSeparator {
+        let mut buffer = [0_u8; 96];
+        buffer[0..32].copy_from_slice(&hex!(
+            "47e79534a245952e8b16893a336b85a3d9ea9fa8c573f3d803afb92a79469218"
```
Comment (fleupold) [inline crates/e2e/src/setup/onchain_components.rs:298]: What are these magic numbers?

**Review by fleupold — APPROVED** [review]
Very cool. This allows for some very nice use cases, where users can e.g. bridge funds to a specific mainnet address and sign an order in the name of that address in which the safe is deployed in a pre-hook and proceeds are swapped back to their address on the original chain in a post hook.

File: crates/e2e/src/setup/onchain_components.rs (hunk 1 above)
Comment (nlordell) [reply crates/e2e/src/setup/onchain_components.rs:298]: Ah sorry - meant to include links to code but forgot. Will update.

Comment (nlordell) [reply crates/e2e/src/setup/onchain_components.rs]: Added some documentation with links to the code that was ported and where the magic numbers come from.

**Review by MartinquaXD — APPROVED** [review]
Nice test and easy to understand. 👍

**Pull request merged**